src/XR/features/WebXRLayers.ts

```diff
@@ -0,0 +1,291 @@
+    textureType: "texture",
+    colorFormat: 0x1908, /* WebGLRenderingContext.RGBA */
+    depthFormat: 0x1902, /* WebGLRenderingContext.DEPTH_COMPONENT */
+    scaleFactor: 1.0,
+};
+
+/**
+ * Exposes the WebXR Layers API.
+ */
+export class WebXRLayers extends WebXRAbstractFeature {
+    /**
+     * The module's name
+     */
+    public static readonly Name = WebXRFeatureName.LAYERS;
+    /**
+     * The (Babylon) version of this module.
+     * This is an integer representing the implementation version.
+     * This number does not correspond to the WebXR specs version
+     */
+    public static readonly Version = 1;
+
+    private _xrWebGLBinding: XRWebGLBinding;
+
+    constructor(_xrSessionManager: WebXRSessionManager) {
+        super(_xrSessionManager);
+        this.xrNativeFeatureName = "layers";
+    }
+
+    /**
+     * Attach this feature.
+     * Will usually be called by the features manager.
+     *
+     * @returns true if successful.
+     */
+    public attach(): boolean {
+        if (!super.attach()) {
+            return false;
+        }
+
+        const glContext = this._xrSessionManager.scene.getEngine()._gl;
+        this._xrWebGLBinding = new XRWebGLBinding(this._xrSessionManager.session, glContext);
+        this.setXRSessionLayers([this.createProjectionLayer()]);
+
+        return true;
+    }
+
+    /**
+     * Creates a new XRWebGLLayer.
+     * @param params an object providing configuration options for the new XRWebGLLayer
+     * @returns the XRWebGLLayer
+     */
+    public createXRWebGLLayer(params = defaultXRWebGLLayerInit): WebXRWebGLLayerWrapper {
+        const glContext = this._xrSessionManager.scene.getEngine()._gl;
+        const layer = new XRWebGLLayer(this._xrSessionManager.session, glContext, params);
+        return new WebXRWebGLLayerWrapper(layer);
+    }
+
+    /**
+     * Creates a new XRProjectionLayer.
+     * @param params an object providing configuration options for the new XRProjectionLayer
+     * @returns the projection layer
+     */
+    public createProjectionLayer(params = defaultXRProjectionLayerInit): WebXRProjectionLayerWrapper {
+        const projLayer = this._xrWebGLBinding.createProjectionLayer(params);
+        return new WebXRProjectionLayerWrapper(projLayer, this._xrWebGLBinding);
+    }
+
+    /**
+     * Sets the layers to be used by the XR session. 
+     * Note that you must call this function with any layers you wish to render to 
+     * since it adds them to the XR session's render state 
+     * (replacing any layers that were added in a previous call to setXRSessionLayers or updateRenderState). 
+     * This method also sets up the session manager's render target texture provider 
+     * as the first layer in the array, which feeds the WebXR camera(s) attached to the session.
+     * @param wrappedLayers An array of WebXRLayerWrapper, usually returned from the WebXRLayers createLayer functions.
+     */
```

Comment:
```suggestion
    /**
     * Sets the layers to be used by the XR session.
     * Note that you must call this function with any layers you wish to render to
     * since it adds them to the XR session's render state
     * (replacing any layers that were added in a previous call to setXRSessionLayers or updateRenderState).
     * This method also sets up the session manager's render target texture provider
     * as the first layer in the array, which feeds the WebXR camera(s) attached to the session.
     * @param wrappedLayers An array of WebXRLayerWrapper, usually returned from the WebXRLayers createLayer functions.
     */
```